Tests/UnitTests/Misc/SDKTesterTests.swift

```diff
@@ -0,0 +1,133 @@
+        let error = ErrorUtils.offlineConnectionError().asPublicError
+        self.purchases.mockedHealthRequestResponse = .failure(error)
+
+        do {
+            try await self.tester.test()
+            fail("Expected error")
+        } catch let SDKTester.Error.failedConectingToAPI(underlyingError) {
+            expect(underlyingError).to(matchError(error))
+        } catch {
+            fail("Unexpected error: \(error)")
+        }
+    }
+
+    func testFailingAuthenticatedRequest() async throws {
+        let error = ErrorUtils
+            .backendError(withBackendCode: .invalidAPIKey, backendMessage: "Invalid API key")
+            .asPublicError
+        self.purchases.mockedCustomerInfoResponse = .failure(error)
+
+        do {
+            try await self.tester.test()
+            fail("Expected error")
+        } catch SDKTester.Error.invalidAPIKey {
+            // Expected error
+        } catch {
+            fail("Unexpected error: \(error)")
+        }
+    }
+
+    func testFailingOfferingsRequest() async throws {
+        let error = OfferingsManager.Error.missingProducts(identifiers: ["a"]).asPublicError
+        self.purchases.mockedOfferingsResponse = .failure(error)
+
+        do {
+            try await self.tester.test()
+            fail("Expected error")
+        } catch let SDKTester.Error.failedFetchingOfferings(offeringsError) {
+            expect(offeringsError).to(matchError(error))
+            expect(self.purchases.invokedGetOfferingsParameters) == .failIfProductsAreMissing
+        } catch {
+            fail("Unexpected error: \(error)")
+        }
+    }
+
+    func testSuccessfulTest() async throws {
+        do {
+            try await self.tester.test()
+        } catch {
+            fail("Unexpected error: \(error)")
+        }
+    }
+
+    // MARK: - Errors
+
+    func testUnknownError() {
+        let underlyingError = ErrorUtils.missingReceiptFileError()
+        let error = SDKTester.Error.unknown(underlyingError)
+
+        expect(error.errorUserInfo[NSUnderlyingErrorKey] as? NSError).to(matchError(underlyingError))
+        expect(error.localizedDescription) == "Unknown error: \(underlyingError.localizedDescription)"
+    }
+
+    func testInvalidAPIKey() {
+        let error = SDKTester.Error.invalidAPIKey
+
+        expect(error.errorUserInfo[NSUnderlyingErrorKey] as? NSNull).toNot(beNil())
+        expect(error.localizedDescription) == "API key is not valid"
+    }
+
+    func testFailedConnectingToAPI() {
+        let underlyingError = OfferingsManager.Error.missingProducts(identifiers: ["a"]).asPublicError
+        let error = SDKTester.Error.failedFetchingOfferings(underlyingError)
+
+        expect(error.errorUserInfo[NSUnderlyingErrorKey] as? NSError).to(matchError(underlyingError))
+        expect(error.localizedDescription) == "Failed fetching offerings: \(underlyingError.localizedDescription)"
+    }
+
+    func testFetchingOfferingsError() {
```

Comment:
And I think this test is also wrongly named

Reply:
Good catch, I messed up all these tests.